replication.py

```diff
@@ -100,10 +101,20 @@ def rule_replicate_batch(ctx, verbose):
                 count_ok += 1
             except msi.Error as e:
                 log.write(ctx, 'ERROR - The file could not be replicated: {}'.format(str(e)))
+
+                # Retry replication with data resource name (covers case where resource is removed from the resource hierarchy).
                 try:
-                    ctx.msi_add_avu('-d', path, errorattr, "{},{}".format(from_path, to_path), "")
-                except Exception:
-                    pass
+                    # Workaround the PREP deadlock issue: Restrict threads to 1.
+                    ofFlags = "numThreads=1++++rescName={}++++destRescName={}++++irodsAdmin=++++verifyChksum=".format(data_resc_name, to_path)
```

Comment:
From an application admin perspective, I think it would be useful to log that the fallback replication behaviour has been triggered, since this should be an unusual situation.